src/test/isolation2/sql/orphan_temp_table.sql

```diff
@@ -0,0 +1,62 @@
+-- Test orphan temp table on coordinator. 
+
+-- case 1: Before the fix, when backend process panic on the segment, the temp table will be left on the coordinator.
+-- create a temp table
+1: CREATE TEMP TABLE test_temp_table_cleanup1(a int);
+
+-- panic on segment 0
+1: SELECT gp_inject_fault('create_function_fail', 'panic', dbid) FROM gp_segment_configuration WHERE role='p' AND content = 0;
+
+-- trigger 'create_function_fail' panic
+1: CREATE FUNCTION my_function() RETURNS void AS $$ BEGIN END; $$ LANGUAGE plpgsql;
+
+-- we should not see the temp table on the coordinator
+1: SELECT oid, relname, relnamespace FROM pg_class where relname = 'test_temp_table_cleanup1';
+-- we should not see the temp namespace on the coordinator
+1: SELECT count(*) FROM pg_namespace where (nspname like '%pg_temp_%' or nspname like '%pg_toast_temp_%') and oid > 16386;
+
+
+-- the temp table is left on segment 0, it should be dropped by autovacuum later
```

Comment:
Autovacuum does not work on 6X!

Reply:
Maybe we should remove the mention of the autovacuum?

Reply:
Maybe use "4. Fix autovacuum issue by using gpcheckcat instead." in this commit instead of last?

Reply:
Done